src/safeposix/syscalls/fs_calls.rs

```diff
@@ -389,41 +389,75 @@ impl Cage {
         }
     }
 
-    //------------------MKNOD SYSCALL------------------
-
+    /// ## ------------------MKNOD SYSCALL------------------
+    /// ### Description
+    /// The `mknod_syscall()` creates a filesystem node (file, device special file or pipe) named by a path as the input parameter.
+    /// The file type and the permissions of the new file are initialized from the "mode" provided as the input parameter.
+    /// There are 5 different file types: S_IFREG, S_IFCHR, S_IFBLK, S_IFIFO, or S_IFSOCK representing a regular file, character special file, block special file, FIFO (named pipe), or UNIX domain socket, respectively.
+    /// The newly created file is empty with size 0.
+    /// On successful completion, the timestamps for both the newly created file and its parent are updated along with their linkcounts.
+    /// 
+    /// ### Function Arguments
+    /// The `mknod_syscall()` receives three arguments:
+    /// * `path` - This argument points to a pathname naming the file.
+    //             For example: "/parentdir/file" represents the new file name as "file", which will be created at this path (/parentdir/file).
+    /// * `mode` - The mode argument specifies both the permissions to use and the type of node to be created. It is a combination (using bitwise OR) of one of the file types and the permissions for the new node.
+    ///            FileType - In LIND, we have only implemented the file type of "Character Device" represented by S_IFCHR flag.
```

Comment:
Don't put more than 4 spaces in a rustdoc comment.  This will cause rustdoc to think it's a test to run.

FYI: @yashaswi2000

Reply:
Fixed.